syncer/syncer.go

```diff
@@ -227,6 +247,13 @@ func (s *Syncer) isClosed() bool {
 
 func (s *Syncer) setATXSynced() {
 	s.atxSyncState.Store(synced)
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	for _, ch := range s.awaitATXSyncedCh {
+		close(ch)
+	}
+	s.awaitATXSyncedCh = make([]chan struct{}, 0)
```

Comment:
Nit,  the `mu` Mutex so far guarded `awaitSyncedCh` only, though it's not documented explicitly what it is meant to be protecting. 😢 

The current implementation might be too strict - `awaitSyncedCh` can be safely accessed while `awaitATXSyncedCh` is modified. How about using a separate Mutex + document what the mutexes guard?